ProjectTaskTracker.js

```diff
@@ -1,77 +1,152 @@
+
+/*----------------------------develop draggable event------------------------------------*/
+const toDoTaskDraggable = new Draggable.Draggable(toDoList, {
+  		draggable: '.tasks'
+	});
+
+const progressTaskDraggable = new Draggable.Draggable(inProgressList, {
+  		draggable: '.tasks'
+	});
 
-// //make Project elements draggable
-// const projectElement = document.getElementsByClassName('sideBar')[0];
-// const projectDraggable =new Draggable.Draggable(projectElement, {
-// 	draggable: '.projects'
-// 	});
+const doneTaskDraggable = new Draggable.Draggable(doneList, {
+  		draggable: '.tasks'
+	});
 
-// //make tasks draggable
-// const taskElement = document.getElementsByClassName('toDoList')[0];
-// const tasksDraggable = new Draggable.Draggable(taskElement, {
-// 	draggable: '.tasks'
-// 	});
+const threeLists = [toDoList, inProgressList, doneList];
 
-const tasks = document.getElementsByClassName('tasks')[0];
-if(tasks){
-	tasks.on('drag:start', ()=> console.log('dragging'));
+let currentMousePositionX = 0;
+let currentMousePositionY = 0;
 
+let currentDraggingItem;
+let dragFromList;
+
+function getDraggingElement(event) {
+	const originalSource = event.data.originalSource;
+	const elementClassName = originalSource.className.split(" ")[0];
+	currentDraggingItem = document.getElementsByClassName(elementClassName)[0];
 }
-// if(tasks){
-// 	for(let task of tasks){
-// 		console.log("here");
-// 		task.on('drag:start', () => console.log('drag:start'));
-// 	}
-// }	
 
+function dragStart(event){
+	//find out which list the element came from and mark down which element is being dragged
+	dragFromListName = (event.sourceContainer.className).split(" ")[1];
+	for(let list of threeLists){
+		let listName = list.className.split(" ")[1];
+		if (listName == dragFromListName){
+			dragFromList = list;
+		}
+	}
+	getDraggingElement(event);
+}
 
-// draggable.on('drag:start', () => console.log('drag:start'));
-// draggable.on('drag:move', () => console.log('drag:move'));
-// draggable.on('drag:stop', () => console.log('drag:stop'));
+function onDragMove(event) {
+	currentMousePositionX = event.sensorEvent.data.clientX;
+	currentMousePositionY = event.sensorEvent.data.clientY;
+}
 
-const toDoList = document.getElementsByClassName('toDoList')[0];
-const inProgressList = document.getElementsByClassName('inProgressList')[0];
-const doneList = document.getElementsByClassName('doneList')[0];
+function isValidDropPosition(list){
+	const listCoordinates = list.getBoundingClientRect();
+	if((listCoordinates.left < currentMousePositionX)
+		 &&(currentMousePositionX < listCoordinates.right) 
+		&&(listCoordinates.top < currentMousePositionY) 
+		&& (currentMousePositionY < listCoordinates.bottom)){
+		return true;
+	}
+	return false;
+}
+
+
+function dragStopMove(event){
+	//find out which list the element is suppose to add to
+	let dragToList;
+
+	for(let list of threeLists){
+		if(isValidDropPosition(list)) {
```

Comment:
see comment above. pass in mouse positions

Reply:
but these mouse positions needs the (drag:on) function to keep updating it, the(drag.stop) event doesn't hold the data